ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py

```diff
@@ -231,6 +232,13 @@ def _compute_static_metrics(
                     if not metric.is_window_metric()
                 ],
             )
+            # Replace NaN values with None
+            logger.warn(
+                "NaN Data Type Not Supported: NaN values will be cast to null in OpenMetadata"
+                " to maintain database parity. Ensure your data handling processes account for this conversion."
+            )
+
+            row = {k: None if math.isnan(v) else v for k, v in dict(row).items()}
```

Comment:
Can this be abstracted to modify any `nan` value submitted in other metrics as well? Maybe this can be applied at a higher level.

Reply:
I will create a decorator that could allow to attach to metric functions ( static, composed, hybrid, etc )

Reply:
found a better place to add it, at the core level that takes care of the same